posthog/session_recordings/queries/session_recording_list_from_filters.py

```diff
@@ -122,17 +99,86 @@ def run(self) -> SessionRecordingQueryResult:
-        return ast.Field(chain=[self._filter.order])
+    def _select(self) -> list[ast.Expr]:
+        return [
+            ast.Alias(alias="session_id", expr=ast.Field(chain=["s", "session_id"])),
+            ast.Call(name="any", args=[ast.Field(chain=["s", "team_id"])]),
+            ast.Call(name="any", args=[ast.Field(chain=["s", "distinct_id"])]),
+            ast.Alias(
+                alias="start_time",
+                expr=ast.Call(name="min", args=[ast.Field(chain=["s", "min_first_timestamp"])]),
+            ),
+            ast.Alias(
+                alias="end_time",
+                expr=ast.Call(name="max", args=[ast.Field(chain=["s", "max_last_timestamp"])]),
+            ),
+            ast.Alias(
+                alias="duration",
+                expr=ast.Call(
+                    name="dateDiff",
+                    args=[ast.Constant(value="SECOND"), ast.Field(chain=["start_time"]), ast.Field(chain=["end_time"])],
+                ),
+            ),
+            ast.Alias(
+                alias="first_url",
+                expr=ast.Call(name="argMinMerge", args=[ast.Field(chain=["s", "first_url"])]),
+            ),
+            ast.Alias(
+                alias="click_count",
+                expr=ast.Call(name="sum", args=[ast.Field(chain=["s", "click_count"])]),
+            ),
+            ast.Alias(
+                alias="keypress_count",
+                expr=ast.Call(name="sum", args=[ast.Field(chain=["s", "keypress_count"])]),
+            ),
+            ast.Alias(
+                alias="mouse_activity_count",
+                expr=ast.Call(name="sum", args=[ast.Field(chain=["s", "mouse_activity_count"])]),
+            ),
+            ast.Alias(
+                alias="active_seconds",
+                expr=ast.Call(
+                    name="divide",
+                    args=[
+                        ast.Call(name="sum", args=[ast.Field(chain=["s", "active_milliseconds"])]),
+                        ast.Constant(value=1000),
+                    ],
+                ),
+            ),
+            ast.Alias(
+                alias="inactive_seconds",
+                expr=ast.Call(name="minus", args=[ast.Field(chain=["duration"]), ast.Field(chain=["active_seconds"])]),
+            ),
+            ast.Alias(
+                alias="console_log_count",
+                expr=ast.Call(name="sum", args=[ast.Field(chain=["s", "console_log_count"])]),
+            ),
+            ast.Alias(
+                alias="console_warn_count",
+                expr=ast.Call(name="sum", args=[ast.Field(chain=["s", "console_warn_count"])]),
+            ),
+            ast.Alias(
+                alias="console_error_count",
+                expr=ast.Call(name="sum", args=[ast.Field(chain=["s", "console_error_count"])]),
+            ),
+        ]
+
+    def _order_by_clause(self) -> ast.OrderExpr:
+        expr = (
+            ast.Call(name="sipHash64", args=[ast.Field(chain=["session_id"])])
```

Comment:
Main addition here @neilkakkar. Idea is that it generates a hash based on the session id and uses that to pick a random sample of the recordings

Reply:
this seems unnecessary 🤔 , unless the worry is around pagination here? You can do `order by rand()` if you just want a random sampling.

Reply:
Yeah, was trying to keep things consistent around pagination, link sharing and repeated runs

Reply:
since the hash is consistent, does this mean the first page after you select random will always be the same? 😅  Again not sure if this is more confusing than it changing every time (and then being able to sort that result list).

I think we can relax the product constraint of pagination working appropriately when random sampling is chosen perhaps.

Reply:
as a user I'd potentially like to be like a gacha, every time I click the toggle I get different recordings

Reply:
i commented elsewhere but i'll add it here so there's one thread :)

i don't think I get random... why?

(it also feels wrong that it's consistently random, if i toggle it on and off i get the same random list each time)

Reply:
let's make sure we get metrics on what people are choosing  / loading ??